### DISCUSSION TITLE
v4.2.6.0

### DISCUSSION THREAD

**Discussion by doug24** [dnGrep/dnGrep#1121] [General]
[![v4.2.6.0](https://img.shields.io/github/downloads/dnGrep/dnGrep/v4.2.6.0/total.svg)](https://github.com/dnGrep/dnGrep/releases/tag/v4.2.6.0)

## Changes in this release:

#849 **Windows 11 context menu**: dnGrep has a new Windows Explorer context menu for Windows 11. This will put 'Search with dnGrep' on the context menu in Windows 11. The original context menu remains and can be run on Windows 10 or Windows 11 (where it appears on the second 'Show more options' context menu).

On Windows 11, the installer will register the new menu, and unregister the original menu.  If you want the old menu on Windows 11, you can add it back in the Options dialog.

On Windows 10, the installer will register the original menu as before.  On either OS, you can choose not to install the context menu using the "Custom' installer option.

The dnGrep Options dialog has options to register or unregister both menus. These checkboxes are disabled when not applicable. 
For the new Windows 11 context menu, you must be running as a regular user (not Administrator) to register or unregister the menu.  Each user on a multi-user system must register the new context menu, you cannot register it for all users.

As before, for the original context menu, you must be running as Administrator to register or unregister the context menu.

For the new menu, the menu text 'Search with dnGrep' has been localized into each of the 23 languages currently available in dnGrep. The language used in the context menu is based on the language and locale of your Windows installation, not the selection in the dnGrep Options dialog. The translations have had minimal testing, so if it is not working for your language, please open an issue.

The 'Search with dnGrep' strings were machine translated, so if you have corrections or improvements, please modify them in [Weblate](https://github.com/dnGrep/dnGrep/wiki/Translate-dnGrep) or open an issue.  

If you are running the portable zip package of dnGrep, you can register the context menu using the Options dialog.  You do not want to register either context menu if you are running from removable media as they modify the registry and Windows app database.

To enable registration of the context menu from the zip package, the files in the zip are now also code signed like the files in the installer package have been.

#864 **Multiple instances start when opening multiple files/folders**: The new Windows 11 context menu supports passing multi-selected folders and files when starting dnGrep. For Windows 10, this is not possible, except to use the singleton instance option so there will only be one copy of dnGrep running at a time.  

#1086 **Running a script from the command line**: The original request was to add an additional command line parameter to set the sort options and sort the results.  This is something that can already be done by running a dnGrep script from the command line, so they have not been added as new command line parameters.  However, when testing this, running a script from the command line would corrupt the window layout, and this bug has been fixed.  You can call a [dnGrep script]( https://github.com/dnGrep/dnGrep/wiki/Scripting-Commands) from the command line using the -script, -sc, or /sc flag with the name of the script.

#1087 **Removing all file extensions from archives or plugin**: Fixed a bug where if you remove all the file extension mappings for the Archives or Plugins in the Options dialog, dnGrep would replace the empty list of extensions with the set of default extensions. Now the mapping list will remain empty.

#1090 **Auto pause**: This issue requests an option to automatically pause the search after the first five matches. There are a couple of changes to the main window to do this: the Search button has been modified to make it a split button with two search variations: `Stop after the first match` and `Pause after the 5 matches`.  The old `Stop after the first match` checkbox has been removed. Click the main part of the Search button for a normal search, or the right side to choose one of the variants. There are new settings in the Options dialog - Search options section to specify how many matches will trigger the stop or pause. Note that because the search is run in multiple threads, it may overrun the setting by one or two matches.

#1096 **Command line arguments with comma in the path**: Fixed a bug processing -folder command line arguments when the path contains commas or semi-colons.  Comma and semi-colon are both used to separate multiple paths, and the parser was not handing this correctly.  

If a path contains spaces, commas, or semi-colons, enclose the path in quotes.  To specify multiple paths, separate them with a comma or semi-colon, and quote each path individually if needed.  You may also include the -folder, -f, or /f flag multiple times on the command line to specify multiple paths, quoting each path if needed.

If you use [Everything](https://www.voidtools.com/) with dnGrep, use the -everything, -e, or /e command line flag to specify the Everything search string.  It must be the last flag on the command line, and dnGrep will take the remainder of the command line verbatim as the Everything search string.

#1101 **Disable autocomplete**: As the issue says, the autocomplete feature sometimes gets in the way. There is a new option enable/disable it in the Options dialog - Search options - Enable autocomplete in text input boxes.

#1112: **New [Bring your own plugin](https://github.com/dnGrep/dnGrep/wiki/Bring-Your-Own-Plugin) application**: [Msg2txt](https://github.com/dnGrep/dnGrep/discussions/1112) to extract text from email messages. Thank-you to @huahua74 for finding this application and helping to work out the details.

**Encoding error when reading extracted text** from external applications such as Tika. When reading extracted text from the external application’s standard output, dnGrep was using the system default encoding, resulting in incorrect characters. The code has been modified to detect the encoding, then read as text using the detected encoding.

### Other updates
This release contains an update to:
 - .NET v8.0.5

**Updated Translations** Updates to several languages, thank-you for your contributions. Many languages need help to complete them. You are welcome and encouraged to complete or improve any of the existing translations, see [Translate dnGrep](https://github.com/dnGrep/dnGrep/wiki/Translate-dnGrep) for more information. Do as much or as little as you like. If you do not want to use Weblate for translations, I will take input in any format and merge the changes. Send me a note in the issues or discussions page.

### Checksum information

Name: dnGREP.4.2.6.x64.msi
SHA-256: c5c347830d3a1feeac35e4cf2fb1efffe3b04ad9f9fba087f8b789eb0fd4f97a

Name: dnGREP.4.2.6.x86.msi
SHA-256: ca0118f8e00c152ae069c8904f48840d4da8bb0cda1871355ad3af9eb76009ea

Name: dnGrep.4.2.6.0.x64.zip
SHA256: dea35ed1462572303d0c549070ca954bfc131adbf6a008616e49b2aaf4643904

Name: dnGrep.4.2.6.0.x86.zip
SHA256: f4b600ba960e8b11b94f5131f30567ae6ad91a66e960671dfabd6e4ee54ac9f1

This program uses free code signing provided by [SignPath.io](https://signpath.io?utm_source=foundation&utm_medium=github&utm_campaign=dngrep), and a certificate by the [SignPath Foundation](https://signpath.org?utm_source=foundation&utm_medium=github&utm_campaign=dngrep).

<hr /><em>This discussion was created from the release <a href='https://github.com/dnGrep/dnGrep/releases/tag/v4.2.6.0'>v4.2.6.0</a>.</em>

**Comment by fade2gray** [comment]
> Windows 11 context menu ...

Just an FYI: This was already functional in v4.1.47 which I'm still using - not only that, it can be bought to the surface by using `Shift+RClick` without the need to select 'Show more options'

**Reply by doug24** [reply]
> Just an FYI: This was already functional in v4.1.47 

No, this is completely new from dnGrep.  It is a new native dll written in c++ and an .msix package installer.

Look at the [two screen shots here](https://github.com/dnGrep/dnGrep/wiki/Start-from-Windows-File-Explorer). The top one is the original dnGrep context menu, and still works.  The next one is the new Windows 11 context menu shown on the main context menu with no need for special `Shift+RClick`.

If you already have a Windows 11 context menu that looks like the second screen shot in the link above, you must be using a third-party application like I described in #849.

**Reply by fade2gray** [reply]
Yes, I appear to have skipped this entirely when reading, "This will put 'Search with dnGrep' on the context menu in Windows 11" - my bad.
